app/controllers/marks_controller.rb

```diff
@@ -0,0 +1,104 @@
+class MarksController < ApplicationController
+  before_action :results, only: [:vimeo_subscribe]
+  before_action :require_spy, only: [:index, :results, :show]
+
+  def twitter
+    Seemore::Application.config.twitter
+  end
+
+  def index
+    @marks = current_spy.marks
+  end
+
+  def results
+    if params[:username].present?
+      search_params = { username: params[:username], id: params[:id] }
+      @search_term = search_params[:username]
+      @vimeo_marks = Mark.vimeo_lookup(@search_term)
+      @twitter_marks = twitter_lookup(@search_term)
+    end
+  end
+
+  def twitter_lookup(search_term)
+    marks = []
+    users = twitter.user_search(search_term)
+    users.each do |user|
+      mark = Mark.create(
+        username: user.screen_name,
+        name: user.name,
+        bio: user.description,
+        link: user.url,
+        image_url: user.profile_image_url(size = :original),
+        uid: user.id,
+        location: user.location,
+        provider: "twitter"
+      )
+      marks.push(mark)
+  
+    end
+    return marks
+  end
+
+  def single_mark_twitter_lookup(search_term)
```

Comment:
This method should either be a private method, or be in another class. The public actions of a controller should correspond directly with controller actions that expose needed variables, call model actions, or route to a view, like the standard CRUD actions and special ones like ```vimeo_subscribe``` and ```twitter_subscribe```.. Since ```single_mark_twitter_lookup()``` is interacting with the Twitter API, it could make sense to not have it be a private method here, but rather in a special lib class that handles interacting with the Twitter API and/or munging Twitter data.